tests/pytests/functional/modules/test_x509_v2.py

```diff
@@ -0,0 +1,1543 @@
+        "signing_cert": ca_cert,
+        "revoked": [],
+    }
+
+
+@pytest.fixture()
+def crl_args_exts():
+    return {
+        "authorityKeyIdentifier": "keyid:always",
+        "issuerAltName": "DNS:salt.ca",
+        "issuingDistributionPoint": {
+            "critical": True,
+            "fullname": [
+                "URI:http://salt.ca/myca.crl",
+            ],
+        },
+        "CRLNumber": 1,
+    }
+
+
+@pytest.fixture()
+def crl_revoked():
+    return [
+        {
+            "serial_number": "01337A",
+            "extensions": {
+                "CRLReason": "unspecified",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "01337B",
+            "extensions": {
+                "CRLReason": "keyCompromise",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "01337C",
+            "extensions": {
+                "CRLReason": "cACompromise",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "01337D",
+            "extensions": {
+                "CRLReason": "affiliationChanged",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "01337E",
+            "extensions": {
+                "CRLReason": "superseded",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "01337F",
+            "extensions": {
+                "CRLReason": "cessationOfOperation",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "013380",
+            "extensions": {
+                "CRLReason": "certificateHold",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "013381",
+            "extensions": {
+                "CRLReason": "privilegeWithdrawn",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "013381",
+            "extensions": {
+                "CRLReason": "aACompromise",
+                "invalidityDate": "2022-11-18 13:37:00",
+            },
+        },
+        {
+            "serial_number": "013382",
+            "extensions": {
+                "CRLReason": "removeFromCRL",
+            },
+        },
+    ]
+
+
+@pytest.mark.parametrize("algo", ["rsa", "ec", "ed25519", "ed448"])
+def test_create_certificate_self_signed(x509, algo, request):
+    privkey = request.getfixturevalue(f"{algo}_privkey")
+    res = x509.create_certificate(signing_private_key=privkey, CN="success")
+    assert res.startswith("-----BEGIN CERTIFICATE-----")
+    cert = _get_cert(res)
+    assert "CN=success" == cert.subject.rfc4514_string()
```

Comment:
```suggestion
    assert cert.subject.rfc4514_string() == "CN=success"
```
Assert what you get against what you expect...